ultralytics/data/explorer/explorer.py

```diff
@@ -466,3 +469,136 @@ def generate_report(self, result):
         TODO
         """
         pass
+
+    def count_labels(self, labels: list):
+        """
+        Count occurrences of given labels in dataset.
+
+        Args:
+            labels (list): List of labels to count their occurrences.
+
+        Returns:
+            filtered_df (pandas.DataFrame): Dataframe containing records containing given labels.
+            label_counts (dict): Dictionary containing labels with their count.
+
+        Example:
+            ```python
+            exp = Explorer()
+            exp.create_embeddings_table()
+            filtered_data, label_counts = exp.count_labels(labels)
+            ```
+        """
+        label_counts = {}
+        df = self.table.to_pandas()
```

Comment:
Avoid turning the entire table to pandas, only convert the needed row to save memory like this - table.to_lance().to_table(columns=["labels"]).to_pandas()